ddtrace/tracer/option.go

```diff
@@ -521,6 +538,15 @@ func WithPropagator(p Propagator) StartOption {
 	}
 }
 
+// WithServiceNamePropagation enables the hashed name of this service to be horizontally
+// propagated to other services (e.g. over HTTP requests). This enhances the new ingestion control page
+// so that it can show which exact service is the root of ingested spans, and therefore cost.
```

Comment:
```suggestion
// propagated to other services (e.g. over HTTP requests). This enhances the ingestion control page
// so that it can show which exact service is the root of ingested spans, and therefore cost.
```
nit: it won't be new after some time passes, and then the docs will just be weird :)

Thank you so much for adding this text. This makes such a big difference to a user! Otherwise a "horizontal service hash propagation" means nothing. Disclaimer: I didn't know either what it meant until this moment!

nit 2: in order to have a prettier and shorter API, does it make sense to call it `WithServicePropagation`? Reason being that we've also pretty much gone with `Service` instead of `ServiceName` (e.g. see `WithService` vs `WithServiceName` which is the older version of it).